### PR TITLE
add make for docs

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

adding single command for building docs

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -104,6 +104,12 @@ Most of the tests in PyTorch Lightning train a trial MNIST model under various t
 
 ## Guidelines
 
+**Hint:**
+we have added basic make doe you can simply execute following actions from project root (only for Unix):
```
Comment (awaelchli) [inline .github/CONTRIBUTING.md:108]:
```suggestion
We have added a basic Makefile so you can simply execute following actions from project root (only for Unix):
```
doe is an animal :)

File: docs/source/conf.py [hunk 2]
```diff
@@ -26,7 +26,7 @@
 sys.path.insert(0, os.path.abspath(PATH_ROOT))
 
 FOLDER_GENERATED = 'generated'
-SPHINX_MOCK_REQUIREMENTS = int(os.environ.get('SPHINX_MOCK_REQUIREMENTS', True))
+SPHINX_MOCK_REQUIREMENTS = int(os.environ.get('SPHINX_MOCK_REQUIREMENTS', 1))
```
Comment (awaelchli) [inline docs/source/conf.py:29]: why is it even an int in the first place lol 
it's only used as a bool :)

File: Makefile [hunk 3]
```diff
@@ -12,8 +15,18 @@ test:
...
 
 	# specific file
-	# python -m coverage run --source pytorch_lightning -m py.test --flake8 --durations=0 -v -k
+	# python -m coverage run --source pytorch_lightning -m pytest --flake8 --durations=0 -v -k
+
+docs:
+	rm -rf ./docs/build
+	rm -rf ./docs/source/generated
+	rm -rf ./docs/source/*/generated
+	rm -rf ./docs/source/api
```
Comment (awaelchli) [inline Makefile:24]: shouldn't this stuff also be in the "clean" section?

**Review by awaelchli — APPROVED** [review]

File: docs/source/conf.py (hunk 2 above)
Comment (Borda) [reply docs/source/conf.py:29]: env variable is string only you canot define bool :P

Comment (awaelchli) [reply docs/source/conf.py:29]: ah right

**Review by s-rog — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
Looks greats !

**Review by rohitgr7 — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Comment by Borda** [comment]
just realized that this shall be in master, but in the end, we want to have it both branches

File: Makefile [hunk 4]
```diff
@@ -1,19 +1,32 @@
...
+.PHONY: test clean docs
+
+# to imitate SLURM set only single node
+export SLURM_LOCALID=0
+# assume you have installed need packages
+export SPHINX_MOCK_REQUIREMENTS=0
 
 test:
+	pip install -r requirements/devel.txt
 	# install APEX, see https://github.com/NVIDIA/apex#linux
```
Comment (carmocca) [inline Makefile:10]: dangling comment

Comment (Borda) [reply Makefile:10]: lets keep it :]

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -104,6 +104,12 @@ Most of the tests in PyTorch Lightning train a trial MNIST model under various t
 
 ## Guidelines
 
+**Hint:**
```
Comment (edenlightning) [inline .github/CONTRIBUTING.md:107]: I would add this under ### Test cases above. 

Also, can you add a link to the boring model as a base test case?

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -104,6 +104,12 @@ Most of the tests in PyTorch Lightning train a trial MNIST model under various t
...
+We have added a basic Makefile so you can simply execute following actions from project root (only for Unix):
```
Comment (edenlightning) [inline .github/CONTRIBUTING.md:108]:
```suggestion
To build the documentation locally, simply execute the following commands from project root (only for Unix):
```

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -104,6 +104,12 @@ Most of the tests in PyTorch Lightning train a trial MNIST model under various t
...
+- clean repo from temp/generated files `make clean`
```
Comment (edenlightning) [inline .github/CONTRIBUTING.md:109]:
```suggestion
- `make clean` (cleans repo from temp/generated files)
```

File: .github/CONTRIBUTING.md [hunk 8]
```diff
@@ -104,6 +104,12 @@ Most of the tests in PyTorch Lightning train a trial MNIST model under various t
...
+- build documentation with `make docs` in _docs/build/html_
```
Comment (edenlightning) [inline .github/CONTRIBUTING.md:110]:
```suggestion
-`make docs` (builds documentation under _docs/build/html_)
```

File: .github/CONTRIBUTING.md (hunk 5 above)
Comment (Borda) [reply .github/CONTRIBUTING.md:107]: this is also for building docs which is not test

File: .github/CONTRIBUTING.md [hunk 9]
```diff
@@ -104,6 +104,12 @@ Most of the tests in PyTorch Lightning train a trial MNIST model under various t
...
+- test the project with `make test`
```
Comment (edenlightning) [inline .github/CONTRIBUTING.md:111]:
```suggestion
- `make test` (run all project's tests)
```

File: .github/CONTRIBUTING.md [hunk 10]
```diff
@@ -104,6 +104,12 @@ Most of the tests in PyTorch Lightning train a trial MNIST model under various t
 
 ## Guidelines
 
+**Hint:**
+To build the documentation locally, simply execute the following commands from project root (only for Unix):
+- `make clean` cleans repo from temp/generated files
+-`make docs` builds documentation under _docs/build/html_
```
Comment (Borda) [inline .github/CONTRIBUTING.md:110]:
```suggestion
- `make docs` builds documentation under _docs/build/html_
```

File: .github/CONTRIBUTING.md (hunk 5 above)
Comment (edenlightning) [inline .github/CONTRIBUTING.md:107]:
```suggestion
### Developments scripts
```

**Review by tchaton — APPROVED** [review]

**Comment by Borda** [comment]
@edenlightning mind re-review/approve? :]

**Review by edenlightning — APPROVED** [review]

**Pull request merged**